src/graph.js

```diff
@@ -0,0 +1,79 @@
+class Graph {
```

Comment:
Can you move your implementation into a separate PR please. Let's only focus on setting up the repo with foundational work here